components/script/dom/webgpu/gpucanvascontext.rs

```diff
@@ -351,33 +362,38 @@ impl GPUCanvasContextMethods<crate::DomTypeHolder> for GPUCanvasContext {
 
     /// <https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-getcurrenttexture>
     fn GetCurrentTexture(&self) -> Fallible<DomRoot<GPUTexture>> {
-        // Step 1
+        // Step 1: If this.[[configuration]] is null, throw an InvalidStateError and return.
         let configuration = self.configuration.borrow();
         let Some(configuration) = configuration.as_ref() else {
             return Err(Error::InvalidState);
         };
-        // Step 2
+        // Step 2: Assert this.[[textureDescriptor]] is not null.
         let texture_descriptor = self.texture_descriptor.borrow();
         let texture_descriptor = texture_descriptor.as_ref().unwrap();
         // Step 6
         let current_texture = if let Some(current_texture) = self.current_texture.get() {
             current_texture
         } else {
-            // Step 3&4
+            // Step 4.1
             self.replace_drawing_buffer();
+            // Step 4.2
             let current_texture = configuration.device.CreateTexture(texture_descriptor)?;
             self.current_texture.set(Some(&current_texture));
+            // We only need to mark new texture
+            self.mark_as_dirty();
             current_texture
         };
-        // Step 5
-        self.mark_as_dirty();
         // Step 6
         Ok(current_texture)
     }
 }
 
 impl Drop for GPUCanvasContext {
     fn drop(&mut self) {
+        if let HTMLCanvasElementOrOffscreenCanvas::HTMLCanvasElement(ref canvas) = self.canvas {
+            let document = document_from_node(&**canvas);
+            document.remove_webgpu_context(self);
```

Comment:
Interacting with DOM objects from Drop implementations is unpredictable (#26488). Consider storing the document's hashmap in a `Rc<RefCell>` so that the canvas context can access it directly instead!

Reply:
Right, this totally slipped my mind.